frontend/amundsen_application/static/js/features/ColumnList/ColumnDetailsPanel/index.spec.tsx

```diff
@@ -0,0 +1,278 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { getColumnLink } from 'utils/navigationUtils';
+import ExpandableUniqueValues from 'features/ExpandableUniqueValues';
+import BadgeList from 'features/BadgeList';
+import ColumnDescEditableText from '../ColumnDescEditableText';
+import ColumnStats from '../ColumnStats';
+import ColumnLineage from '../ColumnLineage';
+import ColumnDetailsPanel, { ColumnDetailsPanelProps } from '.';
+
+const mockColumnDetails = {
+  content: {
+    title: 'column_name',
+    description: 'description',
+    nestedLevel: 0,
+    hasStats: true,
+  },
+  type: { name: 'column_name', database: 'database', type: 'string' },
+  usage: 0,
+  stats: [
+    {
+      end_epoch: 1600473600,
+      start_epoch: 1597881600,
+      stat_type: 'column_usage',
+      stat_val: '111',
+    },
+  ],
+  action: { name: 'column_name', isActionEnabled: true },
+  editText: 'Click to edit description in the data source site',
+  editUrl: 'https://test.datasource.site/table',
+  col_index: 0,
+  index: 0,
+  name: 'column_name',
+  tableParams: {
+    database: 'database',
+    cluster: 'cluster',
+    schema: 'schema',
+    table: 'table',
+  },
+  sort_order: 0,
+  isEditable: true,
+  isExpandable: false,
+  badges: [
+    {
+      badge_name: 'Badge Name 1',
+      category: 'column',
+    },
+  ],
+};
+
+Object.defineProperty(navigator, 'clipboard', {
+  value: { writeText: jest.fn() },
+});
+let mockStats = mockColumnDetails.stats;
+jest.mock('utils/stats', () => ({
+  filterOutUniqueValues: () => mockStats,
+  getUniqueValues: () => mockStats,
+}));
+let mockLineageEnabled = true;
+jest.mock('config/config-utils', () => ({
+  isColumnListLineageEnabled: () => mockLineageEnabled,
+  getMaxLength: jest.fn(),
+}));
+
+describe('ColumnDetailsPanel', () => {
+  const setup = (propOverrides?: Partial<ColumnDetailsPanelProps>) => {
+    const props: ColumnDetailsPanelProps = {
+      columnDetails: mockColumnDetails,
+      togglePanel: jest.fn(),
+      ...propOverrides,
+    };
+    const wrapper = shallow(<ColumnDetailsPanel {...props} />);
+    return { props, wrapper };
+  };
+
+  describe('render', () => {
+    it('renders without issues', () => {
+      expect(() => {
+        setup();
+      }).not.toThrow();
+    });
+
+    it('triggers the panel toggle when the X button is clicked', () => {
+      const { props, wrapper } = setup();
+
+      wrapper.find('.btn-close').simulate('click');
+
+      expect(props.togglePanel).toHaveBeenCalled();
+    });
```

Comment:
This is not a 'render' test, more a 'lifecycle' test, as it involves user actions (clicking). Same for some of the ones below.